packaging/makeself/install-or-update.sh

```diff
@@ -208,20 +219,50 @@ done
 
 # -----------------------------------------------------------------------------
 
-echo "Configure TLS certificate paths"
-if [ ! -L /opt/netdata/etc/ssl ] && [ -d /opt/netdata/etc/ssl ] ; then
-  echo "Preserving existing user configuration for TLS"
-else
+select_system_certs() {
   if [ -d /etc/pki/tls ] ; then
-    echo "Using /etc/pki/tls for TLS configuration and certificates"
+    echo "${1} /etc/pki/tls for TLS configuration and certificates"
     ln -sf /etc/pki/tls /opt/netdata/etc/ssl
   elif [ -d /etc/ssl ] ; then
-    echo "Using /etc/ssl for TLS configuration and certificates"
+    echo "${1} /etc/ssl for TLS configuration and certificates"
     ln -sf /etc/ssl /opt/netdata/etc/ssl
-  else
-    echo "Using bundled TLS configuration and certificates"
-    ln -sf /opt/netdata/share/ssl /opt/netdata/etc/ssl
   fi
+}
+
+select_internal_certs() {
+  echo "Using bundled TLS configuration and certificates"
+  ln -sf /opt/netdata/share/ssl /opt/netdata/etc/ssl
+}
+
+certs_selected() {
+  [ -L /opt/netdata/etc/ssl ] || return 1
+}
+
+test_certs() {
+  /opt/netdata/bin/curl --fail --silent --output /dev/null "${NETDATA_CERT_TEST_URL}" || return 1
```

Comment:
Should we check for 60 exit code only?

```
       60     Peer certificate cannot be authenticated with known CA
              certificates.
```

Reply:
Failing to check at all means we run in to the same potential issue we have now, so I would prefer to just act as if the check failed in that case instead of risking claiming not working.

Reply:
As far as I understand, we only need to use the bundled version when the error is "The peer certificate cannot be authenticated using known CA certificates." Why doing it on any other error? If someone set wrong "test claim URL" or any other not "cant auth using known CA certs" error. Can you provide an example of other error you think we need to use bundle?

Reply:
There are literally hundreds of reasons that the cURL command could fail that have nothing to do with certificates. Any of those cases mean that we _have no idea if the system certificates work or not_.

Unless it’s an offline install though, it’s safer in that case to use our bundled certificates instead of the system certificates, because we _know_ that will work in the common case of trying to connect to the Cloud, while we do not know if the system certificates will.

Reply:
> There are literally hundreds of reasons that the cURL command could fail that have nothing to do with certificates.

That was my point. I don't understand why we need to use bundled for any cURL error if we check for certificate validation.

> because we know that will work in the common case of trying to connect to the Cloud

Then why not do it by default?

---

Anyway, I understand your reasoning, but I don't find it convincing. That is not a blocker from my side.

Reply:
> Then why not do it by default?

Because it isn’t guaranteed to work for any _other_ thing Netdata may need to connect to (think on-prem cloud with a certificate issued by the corporate CA), and it makes security auditing and compliance more difficult for users.

Reply:
Yes, and that is why I think we should do it only on 60 and not on any other error. Also, I think timeout is missing in your curl command.

Reply:
Ping: timeout is missing